tools/opensnoop.py

```diff
@@ -405,6 +419,26 @@ def print_event(cpu, data, size):
 
     printb(b'%s' % event.fname)
 
+    if args.json:
+        eventJ = {
+            "uid": event.uid,
+            "pcomm": event.comm,
+            "pid": event.id & 0xffffffff if args.tid else event.id >> 32,
+            "fd": fd_s,
+            "err": err,
+            "path": event.fname,
+        }
+
+        if args.containersmap:
+            container = containers_map.get_container(event.mntnsid)
+            eventJ["node_name"] = container.NodeName
+            eventJ["pod_name"] = container.PodName
+            eventJ["container_name"] = container.ContainerName
+            eventJ["namespace"] = container.Namespace
```

Comment:
The bcc.containers package could provide a function `enrich_json_with_container`:
```
enrich_json_with_container(eventJ, event.mntnsid)
```

So if we decide to add a new field (such as the container image or the service account), we won't have to patch all tools.

Reply:
Nice idea. Done!